x-pack/plugins/security/server/user_profile/user_profile_service.ts

```diff
@@ -506,11 +506,15 @@ export class UserProfileService {
         this.logger.error(`Privileges check API returned unknown profile UIDs: ${unknownUids}.`);
       }
 
-      // Log profile UIDs for which an error was encountered.
-      if (response.errorUids.length > 0) {
-        this.logger.error(
-          `Privileges check API failed for the following user profiles: ${response.errorUids}.`
-        );
+      // Log profile UIDs and reason for which an error was encountered.
+      if (response.errors?.count) {
+        const uids = Object.keys(response.errors.details);
+
+        for (const uid of uids) {
```

Comment:
tl;dr nothing to change here, but I wanted to make a note.

I am normally hesitant to log entries in a loop like this, as I've encountered performance issues in the past by doing so (not just in this product). That said, we've discussed that this error scenario is a real edge case, and the odds of this getting hit in production is really slim to begin with.